docs/scripts/gen_additional_readmes.py

```diff
@@ -0,0 +1,87 @@
+"""
+This script copies README files into the PrairieLearn documentation.
+It processes the README files from specified source directories, modifies links to point
+to the corresponding files on GitHub, and creates a navigation structure for the documentation.
+"""
+
+import re
+from pathlib import Path
+from types import SimpleNamespace
+
+import mkdocs_gen_files
+
+ROOT = Path.cwd()
+SOURCE_ROOT = "https://github.com/PrairieLearn/PrairieLearn/blob/master/"
+
+# Any links need to be replaced with a link to the file on GitHub
+relative_regex = r"\[(.*?)\]\((\..*?)\)"
+relative_regex_dir = r"\[(.*?)\]\((\..*?\/)\)"
+relative_regex_img = r"!\[(.*?)\]\((\..*?)\)"
+readmes_mapping = [
+    SimpleNamespace(**v) for v in mkdocs_gen_files.config.extra["additional_readmes"]
+]
+doc_site_url = mkdocs_gen_files.config.site_url + "/en/latest/"
+
+
+def build_readme_nav() -> None:
+    nav = mkdocs_gen_files.Nav()
+    robot_notice = """
+!!! note
+    This file was spliced into the documentation.
+""".strip()
```

Comment:
Why should a user care about this? Is this actionable or useful?

Reply:
I added this as a thought that this documentation may be lower quality from an end-user perspective. I would be happy to remove it!

Reply:
"spliced" says nothing to me about quality; and if we think these are lower-quality, they shouldn't be in the docs!

Reply:
Fair point. we should only be splicing files into the documentation that are high quality.

Reply:
The other point here is that the documentation will include text about the "directory", so that robot notice could help clarify that. "This content exists outside the documentation directory, so mentions of the current directory refer to their original location in source code."

Reply:
That's fair. If you want to do something about this, I'd go a step farther and link to the file on GitHub to give context to what "this directory" means. Now that I think about it, the READMEs are pretty useless in isolation (that is, without the Python script in their directory).

Reply:
> I'd go a step farther and link to the file on GitHub to give context to what "this directory" means.

Can you explain this a little more?

Reply:
Yeah! I was thinking of a callout at the top (or bottom?) of the page like the following:

> This documentation was sourced from [`contrib/question_converters/canvas/README.md`](https://github.com/PrairieLearn/PrairieLearn/blob/master/contrib/question_converters/canvas/README.md)

This isn't as explicit as what you were suggesting, but I think it's more generic and more useful, as it allows the user to jump to the directory in question pretty easily.

Reply:
![image](https://github.com/user-attachments/assets/01899a4d-4191-4afd-8bf0-347468bbabac)

Reply:
That looks great!